### ISSUE TITLE
Newline after the semicolons of declaration blocks not work if only one declaration exist

### ISSUE THREAD

**Issue by in-in** [stylelint/stylelint#1835]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Bug
```css
a { top: 0; }
```
I expected warning: `Expected newline after ";"`

> Which rule, if any, is this issue related to?

`"declaration-block-semicolon-newline-after": "always"`

> What CSS is needed to reproduce this issue?

```css
a { top: 0; }
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "declaration-block-semicolon-newline-after": "always"
  }
}

```

> Which version of stylelint are you using?

stylelint@7.1.0

> How are you running stylelint: CLI, PostCSS plugin, Node API?

"node scripts with PostCss"

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

I expected warning: `Expected newline after ";"`

> What actually happened (e.g. what warnings or errors you are getting)?

No warnings

**Comment by jeddy3** [comment]
@in-in Thanks for the report.

I can confirm this behaviour. 

@stylelint/core I'm thinking this might be a documentation thing. 

If we change the behaviour of `declaration-block-semicolon-newline-after` to also check *the last semicolon* in a declaration block then it's going to tread on the toes of the `block-closing-brace-*-before` rules.

Way way back, I recall we resolved another issue this way. We made the `block-opening/closing-brace-*-before/after` rules ignore empty blocks in favour of catching those with `block-no-empty`.

If we do the same thing here, then we can just document this behaviour with "This rules ignores the last semicolon within the declaration block. Use the  `block-closing-brace-*-before` rules to control the whitespace instead."

**Comment by davidtheclark** [comment]
@jeddy3 Your suggestion sounds fine to me.
